src/main/java/com/learning/mfscreener/service/HistoricalNavService.java

```diff
@@ -27,6 +28,7 @@
 import org.springframework.web.util.UriComponentsBuilder;
 
 @Service
+@Loggable
 public class HistoricalNavService {
```

Comment:
@coderabbitai, reduce to cognitive complexity of all methods in this class without loosing functionality